ze_graph_ext.h

```diff
@@ -825,6 +826,112 @@ typedef struct _ze_graph_dditable_ext_1_7_t
 
 } ze_graph_dditable_ext_1_7_t;
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Extension version 1.8
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Device graph properties
+typedef struct _ze_device_graph_properties_3_t
+{
+    ze_structure_type_graph_ext_t stype;                ///< [in] type of this structure
+    void* pNext;                                        ///< [in,out][optional] must be null or a pointer to an extension-specific
+    ze_graph_ext_version_t graphExtensionVersion;       ///< [out] graph extension version
+    ze_graph_compiler_version_info_t compilerVersion;   ///< [out] compiler version
+    ze_graph_format_t graphFormatsSupported;            ///< [out] graph formats supported
+    uint32_t maxOVOpsetVersionSupported;                ///< [out] max OV opset version supported by the compiler
+    ze_graph_version_info_t elfVersion;                 ///< [out] elf container version
+    ze_graph_version_info_t runtimeVersion;             ///< [out] firmware runtime version
+    uint64_t skuVariantId                               ///< [out] SKU variant ID
+
+} ze_device_graph_properties_3_t;
+
+///////////////////////////////////////////////////////////////////////////////
+typedef ze_result_t (ZE_APICALL *ze_pfnDeviceGetGraphProperties_ext_3_t)(
+    ze_device_handle_t hDevice,                             ///< [in] handle of the device
+    ze_device_graph_properties_3_t *pDeviceGraphProperties  ///< [out] query result for graph properties of the device
+    );
+
+///////////////////////////////////////////////////////////////////////////////
+typedef ze_result_t (ZE_APICALL *ze_pfnGraphInitialize_ext_t)(
+    ze_graph_handle_t hGraph                        ///< [in] handle of the graph
+    );
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Stages required to execute the graph
+typedef enum _ze_graph_stages_t
+{
+    ZE_GRAPH_STAGE_COMMAND_LIST_INITIALIZE = 0x1,   ///< Call to pfnAppendGraphInitialize is required
+    ZE_GRAPH_STAGE_INITIALIZE = 0x2                 ///< Call to pfnGraphInitialize is required
+    ZE_GRAPH_STAGE_EXECUTE = 0x4,                   ///< Call to pfnAppendGraphExecute is required
+    ZE_GRAPH_STAGE_FORCE_UINT32 = 0x7fffffff
+
+} ze_graph_stages_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Graph properties
+typedef struct _ze_graph_properties_2_t
+{
+    ze_structure_type_graph_ext_t stype;            ///< [in] type of this structure
+    void* pNext;                                    ///< [in,out][optional] must be null or a pointer to an extension-specific
+    uint32_t numGraphArgs;                          ///< [out] number of graph arguments
+    ze_graph_stages_t stagesRequired;               ///< [out] stages required to create and execute the graph
```

Comment:
Is ze_graph_stages_t suppose to be a flag?

Reply:
It's meant to be a bitmask. I'll switch to unit32_t and update the comment.